include/onnxruntime/core/common/common.h

```diff
@@ -95,6 +99,84 @@ void LogRuntimeError(uint32_t session_id, const common::Status& status, const ch
 #define ORT_WHERE_WITH_STACK \
   ::onnxruntime::CodeLocation(__FILE__, __LINE__, __PRETTY_FUNCTION__, ::onnxruntime::GetStackTrace())
 
+#ifdef ORT_NO_EXCEPTIONS
+
+// clang-format off
+#define ORT_TRY { if (true)
+#define ORT_CATCH(x) else if (false)
+#define ORT_RETHROW
+#define ORT_CATCH_END }
+// clang-format on
+
+// In order to ignore the catch statement when a specific exception (not ... ) is caught and referred
+// in the body of the catch statements, it is necessary to wrap the body of the catch statement into
+// a lambda function. otherwise the exception referred will be undefined and cause build break
+#define ORT_HANDLE_EXCEPTION(func)
+
+// TODO, consider changing the output of the error message from std::cerr to logging when the
+// exceptions are disabled, since using std::cerr might increase binary size, and std::cerr output
+// might not be easily accesible on some systems such as mobile
+
+// Throw an exception with optional message.
+// NOTE: The arguments get streamed into a string via ostringstream::operator<<
+// DO NOT use a printf format string, as that will not work as you expect.
+#define ORT_THROW(...)                                                                       \
+  do {                                                                                       \
+    std::cerr << ::onnxruntime::OnnxRuntimeException(ORT_WHERE_WITH_STACK,                   \
+                                                     ::onnxruntime::MakeString(__VA_ARGS__)) \
+                     .what()                                                                 \
+              << std::endl;                                                                  \
+    abort();                                                                                 \
+  } while (false)
+
+// Just in order to mark things as not implemented. Do not use in final code.
+#define ORT_NOT_IMPLEMENTED(...)                                                                \
+  do {                                                                                          \
+    std::cerr << ::onnxruntime::NotImplementedException(::onnxruntime::MakeString(__VA_ARGS__)) \
+                     .what()                                                                    \
+              << std::endl;                                                                     \
+    abort();                                                                                    \
+  } while (false)
+
+#define ORT_LOGIC_ERROR(what_arg)       \
+  do {                                  \
```

Comment:
I don't think we need this

Reply:
There are tests require to catch logic_error exceptions, that's the reason I added this

---
In reply to: [476242979](https://github.com/microsoft/onnxruntime/pull/4894#discussion_r476242979) [](ancestors = 476242979)